src/plugins/expressions/common/util/create_error.ts

```diff
@@ -32,6 +34,11 @@ export const createError = (err: string | Error | ErrorLike): ExpressionValueErr
         : undefined,
     message: typeof err === 'string' ? err : String(err.message),
     name: typeof err === 'object' ? err.name || 'Error' : 'Error',
-    original: err instanceof Error ? err : undefined,
+    original:
+      err instanceof Error
+        ? err
+        : typeof err === 'object' && 'original' in err && err.original instanceof Error
```

Comment:
I think you could leverage some knowledge from TS here and just do
```
...
: err.original instanceof Error ? err.original : undefined
```

Reply:
This code is necessary because `err` could be a string based on the type, so we have to do some generics checks first, otherwise typescript will complain about accessing `original` on a union type that might not have the property